src/main/java/org/generationcp/middleware/pojos/ims/Transaction.java

```diff
@@ -130,7 +125,7 @@ public Transaction(final Integer id, final Integer userId, final Lot lot, final
 		this.sourceRecordId = sourceRecordId;
 		this.previousAmount = previousAmount;
 		this.personId = personId;
-		this.inventoryID = inventoryID;
+		this.lot.setStockId(inventoryID);
```

Comment:
not sure if this is needed but we should check if lot is null in case that it is nullable in ims_transaction...

Reply:
Ok